tensorboard/plugins/core/core_plugin.py

```diff
@@ -482,6 +482,17 @@ def define_flags(self, parser):
 it receives new data at least once per hour)\
 ''')
 
+    parser.add_argument(
+        '--generic_data',
+        type=lambda s: frozenset(x for x in s.split(',') if x),
+        default=frozenset(),
+        help='''\
+[experimental] Use generic data provider infrastructure for plugins in
```

Comment:
Is there any real need to be able to activate this plugin-by-plugin?  It seems like it will make it more onerous to test (especially as more plugins are converted, since we'll have to remember which ones they are and pass that at the command line) versus having the flag just switch all plugins that support the data provider API to use it.

Reply:
I originally had just `--generic_data=true` as you suggest. But given
that we explicitly plan to stage this as “scalars first, then tensors,
then blob sequences”, I think that it would be nice to be able to flip
defaults individually. Once we’re reasonably confident that the scalars
functionality is working as intended, we can enable that by default (for
aggressive dogfooding) while we start work on the images plugin.

It seems to me like this is a lesson that we can learn from the existing
DB mode, which only has a coarse flag and which is still off by default,
and for which the images plugin just doesn’t work.

I considered having individual `--generic_scalars=true` flags, which we
could operate like Bazel’s `--experimental_*` flags, and which solves
the problem of remembering the whole list, but it seemed a bit involved
to convert all the main plugins to define their own flags in loaders.
I suppose that we could just hack them all into `core_plugin`, and that
wouldn’t be so bad. Thoughts?

Reply:
For the record, in discussion we agreed on using a tri-state `false/auto/true`, defaulting to `auto`, in which `true` forces all plugins with support on, `false` forces them all off (as a failsafe), and `auto` lets each plugin choose whether it's on or off by default.  This way as support within a plugin matures (e.g. for scalars), we can flip it on by default without loosing the ability to both force-disable it and also enable the more experimental plugins for testing.

Reply:
Yep. Done.